packages/clerk-js/src/ui/components/OrganizationProfile/InviteMembersForm.tsx

```diff
@@ -89,24 +89,34 @@ export const InviteMembersForm = (props: InviteMembersFormProps) => {
       .inviteMembers({ emailAddresses: emailAddressField.value.split(','), role: roleField.value as MembershipRole })
       .then(onSuccess)
       .catch(err => {
+        removeInvalidEmails(err);
+
         if (isClerkAPIResponseError(err) && err.errors?.[0]?.code === 'duplicate_record') {
           const unlocalizedEmailsList = err.errors[0].meta?.emailAddresses || [];
 
           // Create a localized list of email addresses
           const localizedList = createListFormat(unlocalizedEmailsList, locale);
           setLocalizedEmails(localizedList);
-
-          // Remove any invalid email address
-          const invalids = new Set(unlocalizedEmailsList);
-          const emails = emailAddressField.value.split(',');
-          emailAddressField.setValue(emails.filter(e => !invalids.has(e)).join(','));
         } else {
           setLocalizedEmails(null);
           handleError(err, [], card.setError);
         }
       });
   };
 
+  const removeInvalidEmails = (err: any) => {
+    if (!isClerkAPIResponseError(err)) {
+      return;
+    }
+
+    const invalidEmails = new Set([
+      ...(err.errors[0].meta?.emailAddresses ?? []),
+      ...(err.errors[0].meta?.identifiers ?? []),
```

Comment:
❓ where is the `meta.identifiers` returned? Do we need this? If so, let's add a test to verify that we handle that.

Reply:
👆

Reply:
It's included in the API error. Well i was aiming to test it as part of the `removes blocked/not allowed emails from input after error` test, but apparently used the old `emailAddresses` parameter. I will update it accordingly